File: 2018 - 1/Estruturas de Dados/Alguns códigos/PilhaDinamica/Pilha.c
#include <stdio.h>
#include "Pilha.h"
#include <stdlib.h>



void Empilha(struct Pilha *p, int elemento){
        struct Node *novo = (struct Node*)malloc(sizeof(struct Node));
        novo->valor = elemento;
        novo->next = p->topo;
        p->topo = novo;

}
int Desempilha(struct Pilha *p){

        int elem;
        struct Node *topoAtual = p->topo;
        p->topo = topoAtual->next;
        elem = topoAtual->valor;
        free(topoAtual); // libera memória
        return elem;

}
void Cria(struct Pilha *p){
    p->topo = NULL;

}
int Vazia(struct Pilha *p){
    return p->topo == NULL;

}
int Cheia(struct Pilha *p){

    return 1;

}

File: 2018 - 1/Estruturas de Dados/Alguns códigos/Pilha.cs
public void Empilha(Pilha *p, char x, boolean DeuCerto){
	
	if(Cheia(p) == true)
	DeuCerto = false;
	else {
		Nodeptr pAux = NewNode();
		pAux-> Info = x;
		pAux-> Next = p.Topo;
		p.Topo = pAux;
		DeuCerto == true;
}

}
public void Desempilha(Pilha *p, char x, boolean DeuCerto){
	
	if(p.Vazia() == true)
	DeuCerto = false;
	else {
		x = p.Topo -> Info;
		Nodeptr pAux;
		pAux = p.Topo;
		p.Topo = p.Topo-> Next;
		DeleteNode(pAux);
		DeuCerto = true;
}

}
public void Cria(Pilha *p){
	p.Topo = null;



}
public boolean Vazia(Pilha *p){
	return (p.Topo == null);

}
public boolean Cheia(Pilha *p){
	return false;


}